Lib/_markupbase.py

```diff
@@ -156,6 +156,7 @@ def parse_marked_section(self, i, report=1):
             # look for MS Office ]> ending
             match= _msmarkedsectionclose.search(rawdata, i+3)
         else:
+            match = None
```

Comment:
I've changed this branch to raise an `AssertionError` in https://github.com/python/cpython/pull/8562. Depending on @ezio-melotti's feedback, I can also change it to `return -1`. In both cases, this PR may not be needed so I'd suggest waiting for https://github.com/python/cpython/pull/8562 to be resolved first.

I think it would be nice to have a test case that hits to this branch.

Reply:
[8562](https://github.com/python/cpython/pull/8562)  has been merged, so I think this can be closed.